index.js

```diff
@@ -299,19 +285,64 @@ module.exports = {
     return tree;
   },
 
-  // holds configuration for this addon
-  _config: null,
-
-  // holds configuration for test environment for this addon
-  _configForTest: null,
-
   // controls if code needed to set CSP header in fastboot
   // is included in build output
   _needsFastBootSupport: null,
 
-  // holds calculated policy string
-  _policyString: null,
+  // holds the run config
+  // It's set in `config` hook and used later
+  _runConfig: null,
+
+  // controls if live reload support is append to given CSP policy or not
+  // may be set to `true` by `serverMiddleware` hook
+  _requiresLiveReloadSupport: false,
+
+  // hold live reload configuration such as hostname, port and if using ssl
+  // if live reload is used
+  _liveReloadConfiguration: null,
+
+  // returns the config for a given environment and delivery method
+  _getConfigFor(environment) {
+    debug(`Calculate configuration for environment ${environment}`);
+
+    const { project } = this;
+    const { ui } = project;
+    const ownConfig = readConfig(project, environment);
+    const runConfig = this._runConfig;
```

Comment:
Haven't researched if the run-time configuration is available through some public API on the `project` or `app`. I'm not a big fan of storing it in `config` hook and using it later. But I didn't wanted to touch to many parts at the same time.